File: packages/keep-awake/ios/Plugin/Classes/Results/IsKeptAwakeResult.swift
import Foundation
import Capacitor

@objc public class IsKeptAwakeResult: NSObject, Result {
    let keptAwake: Bool

    init(keptAwake: Bool) {
        self.keptAwake = keptAwake
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["keptAwake"] = keptAwake
        return result as AnyObject
    }
}

File: packages/keep-awake/android/src/main/java/io/capawesome/capacitorjs/plugins/keepawake/classes/results/IsKeptAwakeResult.java
package io.capawesome.capacitorjs.plugins.keepawake.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.keepawake.interfaces.Result;

public class IsKeptAwakeResult implements Result {

    private final boolean keptAwake;

    public IsKeptAwakeResult(boolean keptAwake) {
        this.keptAwake = keptAwake;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("keptAwake", keptAwake);
        return result;
    }
}
